### ISSUE TITLE
compilation failure in stackage nightly

### ISSUE THREAD

**Issue by bergmark** [mrkkrp/req-conduit#12]
```
Preprocessing library for req-conduit-1.0.0..
Building library for req-conduit-1.0.0..
[1 of 1] Compiling Network.HTTP.Req.Conduit

/var/stackage/work/unpack-dir/unpacked/req-conduit-1.0.0-da13cf7a07ff236eae69344ad512ac611211e6ca7a1e8a75839bade3add3aaa0/Network/HTTP/R
eq/Conduit.hs:127:22: error:
    • Couldn't match type ‘c0’ with ‘C.Producer m0 ByteString’
      Expected: L.BodyReader -> c0
        Actual: L.BodyReader -> C.Producer m0 ByteString
      Cannot instantiate unification variable ‘c0’
      with a type involving polytypes: C.Producer m0 ByteString
    • In the first argument of ‘(.)’, namely ‘bodyReaderSource’
      In the expression: bodyReaderSource . L.responseBody
      In an equation for ‘responseBodySource’:
          responseBodySource = bodyReaderSource . L.responseBody
    |
127 | responseBodySource = bodyReaderSource . L.responseBody
    |                      ^^^^^^^^^^^^^^^^

[...]
```

**Comment by mrkkrp** [comment]
I can't manage to reproduce this. I've created this `stack.yaml` file:

```yaml
resolver: nightly-2021-08-07
packages:
- .

nix:
  packages:
    - zlib
    - bzip2
```

When I run `stack build` it succeeds. AFAIU, this should use the same dependencies as your regular nightly builds.

**Comment by mrkkrp** [comment]
It looks like the problem is not present on master, only in the (very old) `req-conduit-1.0.0`. Perhaps it's time to make a new release.

**Comment by mrkkrp** [comment]
Released `req-conudit-1.0.1`. Going to open a PR to re-enable in Stackage.
